### ISSUE TITLE
Final Report for NumFocus Small Development Grant

### ISSUE THREAD

**Issue by kelle** [astropy/astropy-project#169]
We need a final report for the NumFocus Small Development Grant to report on the `specreduce` work done.

**Comment by jradavenport** [comment]
The NumFOCUS Small Development Grant for the Astropy/Specreduce project enabled a small but substantial step forward in both the design of the project, and bringing existing code into the the repository. The funding from this grant allowed J. Davenport to spend 75 hours of effort on the project, including making their first Pull Requests and contributions to the Astropy project, and participate in a remote sprint to help further design the workflow of the specreduce package.

Prior to the grant, many key modules in the specreduce project had been outlined but never implemented. The major result from this grant was to both bring existing code from the functional PyDIS project into the more general astropy/specreduce package, and to more specifically use the PyDIS project's workflow example to better illustrate each step in a typical user experience for specreduce. 

Davenport developed a detailed workflow exploration notebook, illustrating a typical-user workflow for this package would be. This example fully reduces actual raw data into a science-ready product, using only Python code (i.e. no dependence on the external legacy IRAF project):
https://github.com/jradavenport/specreduce/blob/workflow_exploration/specreduce/notebooks/apo05.ipynb

Additional specific workflow examples were developed to highlight needs for each major component of specreduce that still needs development, including spectroscopic flat-fielding, identify, and trace/extract/flux calibration.
https://github.com/jradavenport/specreduce/blob/workflow_exploration/specreduce/notebooks/flatfield_demo.ipynb
https://github.com/jradavenport/specreduce/blob/workflow_exploration/specreduce/notebooks/identify_demo.ipynb
https://github.com/jradavenport/specreduce/blob/workflow_exploration/specreduce/notebooks/fluxcal_demo.ipynb

These notebooks have highlighted specific differences in the fluxcalbiration that should be included in specreduce, as distinct from the general image combine already developed in astropy/ccdproc
We also have demonstrated a way to run a simple identify in the notebook. However, discussions during the specreduce sprint found that this will be one of the most difficult pieces of the project to develop a single solution for, and so many implementations are likely needed for a variety of specific use cases.

Additionally, during the sprint we started a new repository, the specreduce-data package, which will contain reference and calibration data, both from classical archives that previously have not had formal maintain or provenance (e.g. from IRAF), as well as providing a framework for improved calibration data for future instruments.
The apo05 example data, as well as APO extinction file, will be included as part of that project
https://github.com/astropy/specreduce-data

## PRs with contributions (code or discussion)
- https://github.com/astropy/specreduce/pull/45 % add flux calibration functionality, major addition to specreduce toolkit, directly based on the PyDIS-based workflow
- https://github.com/astropy/specreduce/pull/58 % new specreduce_data package
- https://github.com/astropy/specreduce/pull/59/ % new extinction class

**Comment by eteq** [comment]
Thanks @jradavenport ! This is great.  One quick question: even though eventually we want to implement enough of it in `specreduce` that those notebooks become out-of-date, those notebooks you made are probably useful for someone who *right now* wants to implement a pipeline.  Maybe we should make them a bit more visible, perhaps by linking to them from the specreduce docs?  If we did that the only caveat is that we want to make sure we set up the links so that they're long-lasting - e.g., you might make a branch in your fork called `for-specreduce-docs` and we link to that instead of your primary branch. (Or if you have some other way you think would be good of making those notebooks more visible for `specreduce` users, I'm open to that too!)

(Once we resolve that I think this can be closed!)
